types/mongodb/index.d.ts

```diff
@@ -1387,10 +1387,10 @@ export type PullAllOperator<TSchema> = ({
 export type UpdateQuery<TSchema> = {
     /** https://docs.mongodb.com/manual/reference/operator/update-field/ */
     $currentDate?: OnlyFieldsOfType<TSchema, Date, true | { $type: 'date' | 'timestamp' }>;
-    $inc?: OnlyFieldsOfType<TSchema, number | undefined>;
+    $inc?: OnlyFieldsOfType<TSchema, number | Decimal128 | undefined>;
```

Comment:
We probably should create an alias for numeric types to add here, so that we can use it where needed. Something like `type NumericTypes = number | Decimal128 | Double | Int32 | Long`. (or `QueryNumericTypes`)

Reply:
Agree @CaselIT. Sorry I wasn't able to work on this for awhile, but I addressed this comment in:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48934

Let me know if you agree or want further changes.